### PR TITLE
feat(workflows): adds registry syncer

### PR DESCRIPTION
**Pull request by MStreet3**
Adds workflow registry syncing service with a null implementation.  Service is not wired into the application.  Only handles ForceSecretsUpdateRequests.

### CODE REVIEW

**Review by cedric-cordenier — DISMISSED** [review]

File: core/services/chainlink/application.go [hunk 1]
```diff
@@ -215,7 +216,7 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 
 	// TODO: wire this up to config so we only instantiate it
 	// if a workflow registry address is provided.
-	workflowRegistrySyncer := syncer.NewWorkflowRegistry()
+	workflowRegistrySyncer := syncer.NewWorkflowRegistry[commontypes.ContractReader](nil, nil, nil, nil, "")
```
Comment (MStreet3) [inline core/services/chainlink/application.go]: type inference only needed since we are passing nil at the moment.

**Review by cedric-cordenier — DISMISSED** [review]

File: core/services/workflows/syncer/workflow_registry.go [hunk 2]
```diff
@@ -2,39 +2,552 @@ package syncer
...
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker:
+			// for each event type, send a signal for it to execute a query and produce a new
+			// batch of event logs
+			for i := 0; i < len(w.eventTypes); i++ {
+				select {
+				case signal <- struct{}{}:
```
Comment (ettec) [inline core/services/workflows/syncer/workflow_registry.go:318]: this I think is a bug - there is no guarantee that the signal would not be consumed by just a few queryEvent loops and never seen by any of the other loops (in the extreme case), meaning  you would see a certain event types but not events of other types

**Review by ettec — DISMISSED** [review]

File: core/services/relay/evm/capabilities/workflows/syncer/workflow_syncer_test.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+func Test_SecretsWorker(t *testing.T) {
+	var (
+		ctx       = coretestutils.Context(t)
+		lggr      = logger.TestLogger(t)
+		backendTH = testutils.NewEVMBackendTH(t)
+		db        = pgtest.NewSqlxDB(t)
+		orm       = syncer.NewWorkflowRegistryDS(db, lggr)
+
+		giveTicker     = signalers.MakeTicker(ctx.Done(), 500*time.Millisecond)
```
Comment (jmank88) [inline core/services/relay/evm/capabilities/workflows/syncer/workflow_syncer_test.go]: What does this helper do? Would something go wrong if the ticker was used directly instead?
```suggestion
		ticker     := time.NewTicker(500*time.Millisecond)
		t.Cleanup(ticker.Stop)
```

Comment (MStreet3) [reply core/services/relay/evm/capabilities/workflows/syncer/workflow_syncer_test.go]: the ticker on the worker isn't actually a time ticker, it's a `chan struct{}`.  This helper turns a time interval into a `chan struct{}`.  in smaller unit tests we can just send the structs manually to trigger side effects, but with the log poller as a dependency the worker has to repeatedly check what's been polled.

Comment (jmank88) [reply core/services/relay/evm/capabilities/workflows/syncer/workflow_syncer_test.go]: But why not just use the standard `chan time.Time` ticker channel? Why is it important or beneficial to force a conversion to another type?

**Review by cedric-cordenier — DISMISSED** [review]

File: core/utils/signalers/signalers.go
Comment (jmank88) [inline core/utils/signalers/signalers.go:1]: Please reconsider this helper

**Review by jmank88 — CHANGES_REQUESTED** [review]

File: core/utils/signalers/signalers.go
Comment (MStreet3) [reply core/utils/signalers/signalers.go:1]: removed

**Review by cedric-cordenier — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by ilija42 — APPROVED** [review]

**Pull request merged**